test/e2e/framework/timer/timer.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package timer
+
+import (
+	"time"
+
+	"bytes"
+	"fmt"
+
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/perftype"
+)
+
+var now = time.Now
+
+// Represents a phase of a test. Phases can overlap.
+type Phase struct {
+	sequenceNumber int
```

Comment:
You seem to set this sequence number manually for ordering the steps. However, that doesn't seem very convenient and is an overhead for someone adding new phases to rightly calculate it by looking at ordering of other code blocks. I suggest automating it - probably by using a counter inside TestPhaseTimer.

Reply:
I added a comment to StartPhase explaining why this is necessary.

Reply:
Yup.. saw that. But I'm not sure I understand why that can't be auto-calculated using a counter?

Reply:
Please address.

Reply:
Are you sure you've seen the extended comment at https://github.com/kubernetes/kubernetes/pull/54992/files#diff-798659a13d17841486bbd8ceb70acb64R87? I pushed it only a dozen or so minutes ago, so I don't think that's what you referred to when writing "Yup.. saw that."? Let me know if it's still not clear enough.

Reply:
Nope... I was seeing the older comments when I wrote that:

```
// It ends the implicit "prep" phase if it was still running.
 // sequenceNumber is an integer prepended to phaseName in the output
```

I'm still not convinced that the way to solve the issue with ordering for perfdash is by using such synthetic numbering. The msg should be used as the phase ID instead. Besides, what if we run out of numbers in b/w 2 steps?

Reply:
I agree that it would be best if there was a separate field for sorting, and another to use as a stable label. But currently the `perftype.DataItem` schema does not support that. So this concatenation of 3-digit number and label is a hack to get around that limitation.

It's no big deal if we run out of numbers: we'll just reuse them, they don't **have to** be unique. This hack is only meant to provide rough sorting, such that initialization steps at least show up earlier than shutdown steps. I think it's better than having a complete mess, but if you feel strongly against it, I can remove the numeric prefix.

Reply:
That's alright for now.